ui/ui_statusbar.c

```diff
@@ -13,111 +13,122 @@
 #include "display/scope.h"
 
 
-uint32_t ui_statusbar_info(char *buf)
+uint32_t ui_statusbar_info(char *buf, size_t buffLen)
 {
     uint32_t len=0;
     uint32_t temp=0;
     uint32_t cnt=0;
 
-	len+=ui_term_color_text_background_buf(&buf[len], 0x000000, BP_COLOR_GREY);
+	len+=ui_term_color_text_background_buf(&buf[len], buffLen, 0x000000, BP_COLOR_GREY);
+    buffLen = sizeof(tx_sb_buf) - len;	
     
     if(system_config.psu)
     {
-        temp=sprintf(&buf[len],"Vout: %u.%uV",
+        temp=snprintf(&buf[len], buffLen, "Vout: %u.%uV",
            (system_config.psu_voltage)/10000, ((system_config.psu_voltage)%10000)/100
         );
         len+=temp;
         cnt+=temp;
+        buffLen = sizeof(tx_sb_buf) - len;	
         if(system_config.psu_current_limit_en)
         {
-            temp=sprintf(&buf[len],"/%u.%umA max",
+            temp=snprintf(&buf[len], buffLen, "/%u.%umA max",

```

Comment:
`buffLen - len` here?

Reply:
Yes, that is much more clean.  That way `buffLen` can remain more or less "constant" and represent the full size of the buffer and we just subtract off the running `len` (representing how many chars have already been consumed).